Runtime/MeshCombiner.cs

```diff
@@ -516,6 +516,46 @@ private static Matrix4x4 ScaleMatrix(ref Matrix4x4 matrix, float scale)
                 m33 = matrix.m33 * scale
             };
         }
+
+        static bool CanReadMesh(Mesh mesh) {
+            #if UNITY_EDITOR
+            return CanReadMeshInEditor(mesh);
+            #else
+            return mesh.isReadable;
+            #endif
+        }
+
+#if UNITY_EDITOR
+        static System.Reflection.MethodInfo Mesh_canAccess;
+        // This is a workaround for a Unity peculiarity - 
+        // non-readable meshes are actually always accessible from the Editor.
+        // We're still logging a warning since this won't work in a build.
+        static bool CanReadMeshInEditor(Mesh mesh)
+        {
+            if (mesh.isReadable) return true;
+            if (!Application.isPlaying) return true;
+
+            if (Mesh_canAccess == null)
+                Mesh_canAccess = typeof(Mesh).GetProperty("canAccess", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance).GetMethod;
+
+            if(Mesh_canAccess != null)
+            {
+                try {
```

Comment:
Please put the curly brackets on their own lines to follow the style of the rest of the code.